compiler/z/codegen/S390Peephole.cpp

```diff
@@ -1525,18 +1525,10 @@ bool swapOperands(TR::Register * trueReg, TR::Register * compReg, TR::Instructio
       }
       return true;
    }
-
-void insertLoad(TR::Compilation * comp, TR::CodeGenerator * cg, TR::Instruction * i, TR::Register * r)
+void insertLoad(TR::CodeGenerator * cg, TR::Instruction * i, TR::Register * r)
    {
-   switch(r->getKind())
-     {
-     case TR_FPR:
-       new (comp->trHeapMemory()) TR::S390RRInstruction(TR::InstOpCode::LDR, i->getNode(), r, r, i, cg);
-       break;
-     default:
-       new (comp->trHeapMemory()) TR::S390RRInstruction(TR::InstOpCode::LR, i->getNode(), r, r, i, cg);
-       break;
-     }
+   TR::Compilation * comp = cg->comp();
+   r->getKind() == TR_FPR ? new (comp->trHeapMemory()) TR::S390RRInstruction(TR::InstOpCode::LDR, i->getNode(), r, r, i, cg) : new (comp->trHeapMemory()) TR::S390RRInstruction(TR::InstOpCode::LR, i->getNode(), r, r, i, cg);
```

Comment:
Based on #5141, I think the intention was to have the ternary operator placed on the first argument to the constructor, rather than having two constructor calls and using a ternary operator to decide between them. Additionally, it might be better to use the `generateRRInstruction` helper rather than calling the constructor directly.

```suggestion
   generateRRInstruction(cg, r->getKind() == TR_FPR ? TR::InstOpCode::LDR : TR::InstOpCode::LR, i->getNode(), r, r, i);
```

Reply:
Thanks Ben, didn't think after of this, but this seems like a better approach.
I will add the suggestion to the branch